### PR TITLE
Remove encoding amp

### LINKED ISSUE #7871: Many fields in AMP ATI calls mis-rendered
**Describe the bug**
Since #7775 was released, AMP pages have been escaping variable names (e.g. `${documentReferrer}` as `%24%7BdocumentReferrer%7D` and so they have not been substituted correctly.

Affected fields include `r`, `re`, `hl`, `lng`, `x5`, `x6`, `ref`

**To Reproduce**

1. Go to a Simorgh AMP page
2. View the ATI call
3. See that some of the values are the variable names rather than the content

**Expected behaviour**
The fields should contain the variable values

**Full example**

```
https://a1.api.bbc.co.uk/hit.xiti?s=598342&s2=70&p=news%3A%3Apidgin.news.story.54075850.page&r=%24%7BscreenWidth%7Dx%24%7BscreenHeight%7Dx%24%7BscreenColorDepth%7D&re=%24%7BavailableScreenWidth%7Dx%24%7BavailableScreenHeight%7D&hl=%24%7Btimestamp%7D&lng=%24%7BbrowserLanguage%7D&x1=[urn%3Abbc%3Acps%3Aa5061e71-355e-4fe9-b0d4-d2336b23eed9]&x2=[amp]&x3=[news-pidgin]&x4=[pcm]&x5=[%24%7BsourceUrl%7D]&x6=[%24%7BdocumentReferrer%7D]&x7=[article]&x8=[simorgh]&x9=[%22MSME%2BSurvival%2BFund%2Bregistration%2Bportal%22%2BHow%2Bto%2Bapply%3A%2BPresident%2BBuhari%2Bokay%2BN60bn%2B%26%2BN15bn%2B%22Guaranteed%2BOfftake%2BSchemes%22%2BCovid-19%2Bsupport%2Bmoni%2Bwey%2BFG%2Bwan%2Bgive%2Bsmall%2Bbizness%2Bfor%2BNigeria%2B-%2BBBC%2BNews%2BPidgin]&x11=[2020-09-10T17%3A43%3A09.000Z]&x12=[2020-09-21T13%3A00%3A10.000Z]&x13=[Nigeria~Coronavirus%2Bpandemic~Careers~Economics~Employment]&x14=[3d5d5e30-dd50-4041-96d5-c970b20005b9~5fe79b8d-56e5-4aff-8b05-21f9ad731912~61da8e70-0be5-4cc0-96be-099bc396c960~ca170ae3-99c1-48db-9b67-2866f85e7342~ee4d5541-afdc-4511-9ba6-a42823b19429]&x16=[WS%20-%20Educate%20me]&x17=[News]&ref=%24%7BdocumentReferrer%7D
```

### PR DESCRIPTION
**Pull request by andrewscfc**
Resolves https://github.com/bbc/simorgh/issues/7871

**Overall change:**
No longer encodes any ati variables that use [amp variable substitutions](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md). 

1. Create index.html file inside simorgh/public folder with the following contents:
```
<html>
  <body>
    <a href="http://localhost:7080/mundo.amp">mundo</a>
  </body>
</html>
```

2. Run [http-server](https://www.npmjs.com/package/http-server) in simorgh directory
3. Visit http://127.0.0.1:8080
4. Click the mundo link
5. Observe hit in in ATI inspector:

Before (various variables not being substitued by amp: `$(sourceUrl)`, `$(screenwidth)`):
image

After (correctly substitued values):
image

The referrer value remains erroneously encoded due to outstanding issue found here: https://github.com/bbc/simorgh/pull/7854#issuecomment-696599841

**Code changes:**
- set `disableEncoding` on all parameters using amp variable substitution on the amp platform only
- Attempt to clear up logic where escaping rules seem arbitrary
  - Add disableEncodingDueToAmpSubstitution var and supporting comment: https://github.com/bbc/simorgh/compare/remove-encoding-amp?expand=1#diff-fe71df61239886a9072d7f8ae0316cb5R51
  - Double encode in one place to make it clearer where the encoding is happening: https://github.com/bbc/simorgh/compare/remove-encoding-amp?expand=1#diff-fe71df61239886a9072d7f8ae0316cb5R57

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

Please ensure this is tested on both amp and canonical.

### CODE REVIEW

File: src/app/containers/ATIAnalytics/atiUrl/index.js [hunk 1]
```diff
@@ -46,13 +46,18 @@ export const buildATIPageTrackPath = ({
...
-  const x6Value =
-    platform === 'amp'
-      ? referrer
-      : referrer && encodeURIComponent(encodeURIComponent(referrer));
+  const x5Value = disableEncodingDueToAmpSubstitution
+    ? href
+    : href && encodeURIComponent(encodeURIComponent(href));
+  const x6Value = disableEncodingDueToAmpSubstitution
+    ? referrer
+    : referrer && encodeURIComponent(encodeURIComponent(referrer));
```
Comment (karinathomasbbc) [inline src/app/containers/ATIAnalytics/atiUrl/index.js:60]: Per my comment https://github.com/bbc/simorgh/pull/7854#discussion_r492602712 - could we check for null values within `encodeURIComponent`?

**Review by karinathomasbbc — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]

File: src/app/containers/ATIAnalytics/atiUrl/index.js (hunk 1 above)
Comment (andrewscfc) [reply src/app/containers/ATIAnalytics/atiUrl/index.js:60]: I just checked this in dev tools:
image

Passing in null to `encodeURIComponent` gives the string `'null'`, I don't think we want that? The current code would return `null` as the value and the `x6` parameter would be omitted from the ATI hit which is correct behaviour.

Comment (karinathomasbbc) [reply src/app/containers/ATIAnalytics/atiUrl/index.js:60]: Forgot encodeURIComponent was a built-in function. Thanks.

**Comment by paruchurisilpa** [comment]
LGTM, apart from the outstanding issue mentioned in https://github.com/bbc/simorgh/pull/7854#issuecomment-696599841.

Will raise a bug ticket for that separately so that we have it tracked.

**Pull request merged**